### PR TITLE
Review stake pool worker creation

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#713 

# Overview

- [x] I have slightly re-organized the code around the declaration of the stake pool layer
- [x] I have namespaced the stake pool worker
- [x] I have mounted some simple error handling on the stake pool worker thread

# Comments

### CODE REVIEW

File: lib/core/src/Cardano/Pool/Metrics.hs [hunk 1]
```diff
@@ -172,11 +172,11 @@ newStakePoolLayer db nl tr = do
...
 -- instead of 'IO'.
 monitorStakePools
-    :: NetworkLayer IO t Block
+    :: Trace IO Text
+    -> NetworkLayer IO t Block
     -> DBLayer IO
-    -> Trace IO Text
     -> IO ()
-monitorStakePools nl db tr = do
+monitorStakePools tr nl db = do
```
Comment (KtorZ) [inline lib/core/src/Cardano/Pool/Metrics.hs:179]: Consistency with the rest of the code base, we usually pass the trace as first argument basically everywhere.

**Review by Anviking — APPROVED** [review]
👍

**Review by paweljakubas — APPROVED** [review]

**Comment by KtorZ** [comment]
@anviking are you trying to confuse bors :')?

**Comment by Anviking** [comment]
No :) It had been "in queue" for several hours so I tried to restart it

**Comment by KtorZ** [comment]
Haha, okay ^.^

**Pull request merged**